### PR TITLE
Test Isolation 2 (TBC)

### PR DESCRIPTION
**Pull request by AntoinePrv**
Follow up on #1791

### CODE REVIEW

**Comment by AntoinePrv** [comment]
The Windows tests are fixed, I disable cleaning external `CONDA_PREFIX` from the `PATH` for that platform. Note that it was not previously done, so this is still an improvement for the the Linux tests rather that a deterioration of the Windows tests.

**Comment by AntoinePrv** [comment]
Great, this is looking better! Actually the assertion that is failing was not tested before (there was not deactivation test).
```python
            stdout, stderr = call(["micromamba deactivate"] + evars)
            res = TestActivation.to_dict(stdout)
Error>      assert find_path_in_str(tmp_root_prefix / "condabin", res["PATH"])
```
On Windows, `PATH` does not contain (actually) any path with `condabin` in it after `micromamba deactivate`.
@wolfv is that expected? Or is something missing?

**Comment by wolfv** [comment]
Hmm actually that is curious! I think we should keep `condabin`, even on Windows. Would you be able to rebase the test cleanup? I just changed (and aligned with conda) how activation works, so it might accidentally fix the test failure :)

File: micromamba/tests/test_activation.py [hunk 1]
```diff
@@ -646,75 +576,50 @@ def test_activation(
...
+            assert find_path_in_str(tmp_root_prefix / "condabin", res["PATH"])
+            assert not find_path_in_str(tmp_root_prefix / "bin", res["PATH"])
+            assert find_path_in_str(tmp_root_prefix / "envs" / "xyz", res["PATH"])
+            assert not find_path_in_str(tmp_root_prefix / "envs" / "abc", res["PATH"])
 
             s = [
                 "micromamba activate",
                 "micromamba activate abc",
-                "micromamba activate xyz --stack",
+                "micromamba activate --stack xyz",
```
Comment (wolfv) [inline micromamba/tests/test_activation.py:587]: are we not testing `--stack xyz` vs `xyz --stack` anymore?

**Comment by wolfv** [comment]
Now it's a different but related failure :D

File: micromamba/tests/test_activation.py (hunk 1 above)
Comment (AntoinePrv) [reply micromamba/tests/test_activation.py:587]: We were not testing both. We were testing one, now it;'s the other.
Did not need to be changed though...

Comment (wolfv) [reply micromamba/tests/test_activation.py:587]: Now we're testing both

**Pull request merged**